src/main/java/com/exasol/glue/query/AbstractQueryGenerator.java

```diff
@@ -33,6 +36,16 @@ public String getIdentifier() {
                 + "'\n";
     }
 
+    /**
+     * Removes {@code IDENTIFIED BY} part of the Exasol SQL statement that contains credentials.
+     *
+     * @param input input query string
+     * @return identifier part removed query string
+     */
+    public static String identifierRemoved(final String input) {
```

Comment:
this is not 100% safe. If for example a username or a password contains identified by, it will be filterd.
Not sure if that would be a problem here.